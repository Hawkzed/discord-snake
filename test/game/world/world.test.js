/* eslint-env jest */
const { World } = require('../../../lib/game/world');

describe('World', () => {
  it('generates a world of the correct size', () => {
    const width = 10;
    const height = 10;

    const world = new World(width, height);

    expect(world.height).toBe(height);
    expect(world.width).toBe(width);
  });

  it('renders the world to a string when toString is called', () => {
    const world = new World(10, 10);

    expect(world.toString()).toMatchSnapshot();
  });

  it('returns a char when getCharAt is called', () => {
    const world = new World(10, 10);

    expect(world.getCharAt(0, 0)).toBe('ðŸ”²');
  });
});
