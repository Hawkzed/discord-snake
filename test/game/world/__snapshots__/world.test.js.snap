// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`World renders the world to a string when toString is called 1`] = `
"🔲🔲🔲🔲🔲🔲🔲🔲🔲🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳‍‍🔳🔲
🔲🔲🔲🔲🔲🔲🔲🔲🔲🔲
"
`;
